src/Api/SecretsManager/Controllers/AccessPoliciesController.cs

```diff
@@ -48,21 +76,84 @@ public async Task<ProjectAccessPoliciesResponseModel> GetProjectAccessPoliciesAs
     public async Task<BaseAccessPolicyResponseModel> UpdateAccessPolicyAsync([FromRoute] Guid id,
         [FromBody] AccessPolicyUpdateRequest request)
     {
-        var result = await _updateAccessPolicyCommand.UpdateAsync(id, request.Read, request.Write);
+        var userId = _userService.GetProperUserId(User).Value;
+        var result = await _updateAccessPolicyCommand.UpdateAsync(id, request.Read, request.Write, userId);
 
         return result switch
         {
             UserProjectAccessPolicy accessPolicy => new UserProjectAccessPolicyResponseModel(accessPolicy),
             GroupProjectAccessPolicy accessPolicy => new GroupProjectAccessPolicyResponseModel(accessPolicy),
             ServiceAccountProjectAccessPolicy accessPolicy => new ServiceAccountProjectAccessPolicyResponseModel(
                 accessPolicy),
-            _ => throw new ArgumentException("Unsupported access policy type provided.")
+            _ => throw new ArgumentException("Unsupported access policy type provided."),
         };
     }
 
     [HttpDelete("{id}")]
     public async Task DeleteAccessPolicyAsync([FromRoute] Guid id)
     {
-        await _deleteAccessPolicyCommand.DeleteAsync(id);
+        var userId = _userService.GetProperUserId(User).Value;
+        await _deleteAccessPolicyCommand.DeleteAsync(id, userId);
+    }
+
+    [HttpGet("/projects/{id}/access-policies/people/potential-grantees")]
```

Comment:
Are these related to projects somehow? Wondering if we should make two generic endpoints that we re-use for all entities?

Reply:
They're related to projects in the sense that we check if the user has write permissions to the project to fetch the potential grantees. 

Another option would be 

`[HttpGet("/organizations/{id}/access-policies/people/potential-grantees")]`
`[HttpGet("/organizations/{id}/access-policies/service-accounts/potential-grantees")]`

and just check that the user has secrets manager access. 

The reason I went for this approach was because the user endpoint we were using previously was more restrictive. 

Thoughts on allowing any secret's manager user in the org to hit the endpoint? 

To note the service account potential grantees, fetch checks that the user has write permission to each service account returned.

Reply:
We could check if the user has write access to any project or service account. But not sure if it's worth it.

Reply:
Resolved in https://github.com/bitwarden/server/pull/2628/commits/cc6f876b03a420ebe1f0f44d135825de6f900dab